### PR TITLE
Ubuntu configure and compilation fixes

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
apparently default Ubuntu shell (/bin/sh) does not like it

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11525/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 01-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/00d3ce52b1ae7b5a20a60daaf1b593694b260eba/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/f7e423642d234c01e1626cf9388d7f57b1de9d94/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/d2385bb0a01a90f1291ffb8ab3f9045bbb25ef74/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/58df5b6997cd3b6ab0e2d3b2a69fe5e53c879130/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/1a917bfc796e5745446834a9f3f299d510f179c5/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/6f5627dfb28326d17e3aad9001a1ca4d7ace65e7/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/0a9fff013e1b61d5ab67db878cfa164fb9031ca0/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/816a4b7d8505b7ef4a61cfdbd0c47f288e3eecba/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/3b757e0b051695817773e735e9bb0038cf10ff0e/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/b61ec6a7a57aef4548437bc1a3b2167ba1cac472/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/b70118dfb2232fab188e5ad61a0cd7eaf22ef19b/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/dd2658f7cc44ae0ac0065324455a14478ec5e202/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/7cbb1b8e1af04fa34e09378f5ddb3a8919d3acef/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/838cf021b697f2db9a3f9a55889de3455efa82de/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/0dd6ae47d96400592c30088ec3ecc5921529f087/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/7c056035e270cdc4ee1e433f98b83b4f987bc40c/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/7bf2b5e309d70630b8dbd7d010b5f8ca931f815c/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/7476b379bd3cc70ab2f391121869edb6e879d438/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/3ed7d88d68845ad987bd70f6f1bb29fd88c5b53e/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/45c82b748e314b760447475ee6ca6573ac26c9a5/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/751fed1f0590293ac8cc06148553cccafb86a388/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/920d35929d6572ad67679735a438c40d15aa0a68/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/b8083c12e4f45c2f26fb50cf603ae8a0d074b96b/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/86940da84607bfda6280f103efd42ed6372aaa93/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/d2d78e84d80f5aecc713f8a90c0080289b882869/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/a20235deb777b2702a7ff119fe6c15a6158ee17d/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/e3a1754c24f4a4231b294e1256a16c39adb196fe/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/9f5d83f5d5c0ec6ea56fe695ac5e4b3ae6f04a82/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c0669dacad4be17bbb08bba62d729200/raw/0710783861a2b92ed00c28613231c0fc90400bf1/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/tools/profile/read_profile.c [hunk 1]
```diff
@@ -259,7 +259,7 @@ static const char* thread_list_str(const int *thread_list, char *buf, size_t max
         p += strlen(p);
     }
 
-    if (p >= buf - 1) {
+    if (p >= buf + 1) {
```
Comment (brminich) [inline src/tools/profile/read_profile.c]: can be `p > buf`

**Comment by shamisp** [comment]
What ubuntu version ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11544/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 03-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/50118b1fc181fd6e8fe99cff16d43de4e302fc59/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/29cfbe04424e05aa1b2481798138aac5ffddd49e/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/c2bd2da7983f6aba9b6282bcd76782f9b204bbf2/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/02588f1e2f788d4be9c0ad35922d47d5f17d5e49/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/8d77fb6dceb612a50386bbc45bc9a92b96ba9232/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/419a40e436e21bd3814bde495a5aa55da577cdf7/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/f04725ac01a859c890cb0a925373fd5ffb255dd5/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/c8adb2423249950df5d24d2824872f08295eaa74/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/16129be665593c26a5e4b2150a0ed2cdf23d3f8f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/29f9ce592612ba68e3e8c0e3eb532d628499ca7d/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/144f920dfc165cc19d9aefb17a6eb8305e90e2c6/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/99e92171ac4f4bf5948620a56aa1a0727781bcf4/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/0e1f18ecb077b3da8e97cbad9313ae7a1df75fda/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/6fc4310fdcdbeaebda918f7828dbce298b7fd61a/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/ec2095f5e37e00049aacb4c08ce800bd1293b20b/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/3fb7b812ded4ead7ccdd639c5f0ffe30b29a85de/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/d69494f874bb7e08fd2280e32b67f201a8636986/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/d579535567bba04acd7ea843a92f12f89a109d81/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/1fcb56fa5d9db06b0804791498cbc3f7cb9d8b4f/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/11d1ae1395a841496bf19bb200a872ddd6be61ea/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/4378a5290ac5d7d67da9b69e713fd6d4b6c6ca39/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/14b80b501765bde1cdc315fb54816c3367df07d0/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/d9087184d41c7d0534e1604d7f3fe4f0ec831571/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/cd34f71fb889e5cd3443d6eba2ee8d9ca32c8783/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/362054ee94022ef5ec2cd88e7917c3db59c82cd6/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/83ac8f7b6395e685c088ff084bb010ab26edae3b/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/6cfa3e7dce1cf3b8ca395b75c39653a46447d684/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/98d9de19f5ad27e52ea6f1febcb9abaf3314291f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ae1a65302650e359d2d294898900bb3b/raw/1ce6c8ac2b949c99f741f0289c2289e138ee5f94/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
ref not a tree
bot:bgate:retest

**Comment by yosefe** [comment]
@shamisp ubuntu 18.04.2

**Pull request merged**